lib/c-hyper.c

```diff
@@ -357,15 +357,17 @@ CURLcode Curl_hyper_stream(struct Curl_easy *data,
     }
   }
 
-  if(select_res & CURL_CSELECT_IN) {
-    if(h->read_waker)
-      hyper_waker_wake(h->read_waker);
-    h->read_waker = NULL;
-  }
-  if(select_res & CURL_CSELECT_OUT) {
-    if(h->write_waker)
-      hyper_waker_wake(h->write_waker);
-    h->write_waker = NULL;
+  if (select_res) {
+      if (CURL_CSELECT_IN) {
+          if (h->read_waker)
+              hyper_waker_wake(h->read_waker);
+          h->read_waker = NULL;
+      }
+      if (CURL_CSELECT_OUT) {
+          if (h->write_waker)
+              hyper_waker_wake(h->write_waker);
+          h->write_waker = NULL;
+      }
```

Comment:
1. `make checksrc` will tell you how this breaks the code style
2. You removed the ands from the conditions so this code is now broken!